### ISSUE TITLE
yenv: accept variable from stdin

### ISSUE THREAD

**Issue by t214c** [malxau/yori#114]
Hi, Malcolm.

I looked a lot and I couldn't find a tool that accepts standard input and saves it to a variable. Something like this:

`echo string | yenv var`
with the result:
` echo %var%` > ` string `

This is very useful because currently you have to make a full for loop to do this.

What do you think?

**Comment by malxau** [comment]
I think I'm not understanding correctly :)

Isn't this
`set var=$(echo string)` ?

That's really capturing stdout of echo into the variable.

Readline.exe exists to read from the console and write to standard output, although note it's a direct console read, not a standard input read.  A standard input read seemed redundant here, since it would just be reading from input and writing to output, like `type`.  But if you wanted to set a variable to a user supplied value, it could be:

`set var=$(readline)`

**Comment by t214c** [comment]
>  set var=$(echo string)
Yeah, but this works only with Yori. I was talking about a tool that could work from any batch file.

**Comment by malxau** [comment]
Okay, I think I understand.  I pushed a4a01f070cd0338be4db1223a52ebf8a7faae230 for this, although am still not happy; I think it should also support both a variable and value from stdin, but having done that, it could have a list of variables and values, and...

**Comment by t214c** [comment]
Thanks a lot!

>although am still not happy

Like I said, it doesn't have to be `yenv` at all. I just wanted a simple tool to abstract away the for loop (and character escaping) required to set a command output to a variable by piping. Advanced or multiple variable support is cherry on the top.
